### PR TITLE
Update-AzVmss bug when using both repairAction and Enabled not getting updated

### PR DESCRIPTION
**Pull request by Sandido**
## Description
Only internal business logic was changed.

A bug was reported for Update-AzVmss where trying to update both the RepairAction and EnableAutomaticRepair value in the same call would result in the Enable value not getting updated. 
This is due to a bug where the AutoamticRepairsAction was accidentally added to the VmssUpdate object instead of the Vmss object. These different objects are used for Put or Patch separately. Using both in one operation caused the logic to break down and only 1 value was sent to the vmss to be updated. 

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Compute/Compute/ChangeLog.md [hunk 1]
```diff
@@ -28,6 +28,7 @@
   The non-versioned image aliases are being updated to versioned values in October 2023, and this message is to help urge customers to use the newer versioned image alias values.
 * Changed the installation behavior for the `GuestAttestation` extension in `New-AzVM` and `New-AzVmss` to set the property `EnableAutomaticUpgrade` to true.
 * Changes to `Set-AzVMOperatingSystem` to correct unnecessary mandatory parameters.
+* Fixes the `Update-AzVmss` cmdlet so the `AutomaticRepairGracePeriod`, `AutomaticRepairAction`, and `EnableAutomaticRepair` parameters function correctly.
```
Comment (vidai-msft) [inline src/Compute/Compute/ChangeLog.md:31]:
```suggestion
* Fixed the `Update-AzVmss` cmdlet so the `AutomaticRepairGracePeriod`, `AutomaticRepairAction`, and `EnableAutomaticRepair` parameters function correctly.
```

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs [hunk 2]
```diff
@@ -123,6 +123,17 @@ public override void ExecuteCmdlet()
...
+        [Parameter(
+            Mandatory = false)]
+        public bool EnableAutomaticRepair { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Type of repair action (replace, restart, reimage) that will be used for repairing unhealthy virtual machines in the scale set. Default value is replace.",
+            ValueFromPipelineByPropertyName = true)]
+        [PSArgumentCompleter("Replace", "Restart", "Reimage")]
+        public string AutomaticRepairAction { get; set; }
```
Comment (vidai-msft) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs:135]: Please use ValidateSet attribute for input validation

**Review by vidai-msft — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs (hunk 2 above)
Comment (Sandido) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs:136]: @vidai-msft , added the ValidateSet functionality.

**Comment by vidai-msft** [comment]
@Sandido if "Replace", "Restart", "Reimage" are always the only valid values for `AutomaticRepairAction`, can you please suppress the issue detected by static analysis? Otherwise, it would be considered as breaking change.

**Comment by Sandido** [comment]
> @Sandido if "Replace", "Restart", "Reimage" are always the only valid values for `AutomaticRepairAction`, can you please suppress the issue detected by static analysis? Otherwise, it would be considered as breaking change.

@vidai-msft , I can't seem to find the breaking change suppression file for Compute as indicated by this: https://github.com/Azure/azure-powershell/blob/main/documentation/Debugging-StaticAnalysis-Errors.md#breaking-changes

**Review by vidai-msft — APPROVED** [review]
LGTM

**Pull request merged**